pulser-core/pulser/sequence/sequence.py

```diff
@@ -423,63 +420,63 @@ def switch_device(
                 # Find if there is phase change between pulses or not
                 phase_is_constant = True
                 if ch_sample_phase.size != 0:
-                    phase_is_constant = bool(
-                        np.all(ch_sample_phase == ch_sample_phase[0])
-                    )
-                # Phase_jump_time and cloc_period check
+                    phase_is_constant = bool(np.all(
+                        ch_sample_phase == ch_sample_phase[0]
+                    ))
+                # Phase_jump_time and clock_period check
                 phase_jump_time_check = (
                     o_d_ch_obj.phase_jump_time == n_d_ch_obj.phase_jump_time
                 )
                 clock_period_check = (
                     o_d_ch_obj.clock_period % n_d_ch_obj.clock_period == 0
                 )
-                if phase_is_constant:
-                    if clock_period_check:
-                        channel_match[o_d_ch_name] = n_d_ch_id
-                        channel_match["alert_phase_jump"] = (
-                            phase_jump_time_check is False
-                        )
-                        break
-                    else:
-                        channel_match[o_d_ch_name] = None
-                        channel_match["strict_error_message"] = (
-                            "No channel with phase_jump_time"
-                            + " & clock_period match."
-                        )
-                    continue
-                if phase_jump_time_check and clock_period_check:
+                if clock_period_check and (
+                    phase_is_constant or phase_jump_time_check
+                ):
                     channel_match[o_d_ch_name] = n_d_ch_id
+                    alert_phase_jump = not phase_jump_time_check
                     break
                 else:
                     channel_match[o_d_ch_name] = None
-                    channel_match["strict_error_message"] = (
+                    strict_error_message = (
                         "No channel with phase_jump_time"
                         + " & clock_period match."
                     )
 
         if None in channel_match.values():
-            if channel_match["strict_error_message"] != "":
-                raise ValueError(channel_match["strict_error_message"])
+            if strict_error_message != "":
```

Comment:
A non-empty string evaluates to `True`, so the pyrhonic way to write this is actually
```suggestion
            if strict_error_message:
```